receiver/dockerstatsreceiver/receiver_test.go

```diff
@@ -169,32 +181,52 @@ func TestScrapeV2(t *testing.T) {
 	assert.NoError(t, err)
 	defer twoContainerEngineMock.Close()
 
+	singleContainerWithOptionalResourceAttributesEngineMock, err := dockerMockServer(&map[string]string{
```

Comment:
You may have some refactoring do if https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21111 gets merged before this, since it changed the test case structure slightly. We can cross that bridge when we come to it though

Reply:
Sure, when either this PR or https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21111  is merged, the other one will have to be rebased and apply any necessary changes.